### PR TITLE
Ignore Iceberg Tables

### PR DESCRIPTION
**Pull request by HamzaJugon**
Fixes #

### CODE REVIEW

File: beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/hive/HiveClient.java [hunk 1]
```diff
@@ -126,6 +127,27 @@ public Map<String, String> getTableProperties(String databaseName, String tableN
     }
   }
 
+  // Method to retrieve properties from the Storage Descriptor
```
[the remark this answers is not in the document]
Comment (jmnunezizu) [reply beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/hive/HiveClient.java:130]: I see the same style in the other classes you changed. Please remove them. The code should be clear enough.

File: beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/handler/ExpiredMetadataHandler.java [hunk 2]
```diff
@@ -146,6 +158,30 @@ private boolean cleanupPartition(
...
   }
 
+  private boolean isIcebergTable(CleanerClient client, String databaseName, String tableName) {
+    try {
+      Map<String, String> tableProperties = client.getTableProperties(databaseName, tableName); // fetching table props
+      String tableType = tableProperties != null ? tableProperties.get("table_type") : null;
+      String format = tableProperties != null ? tableProperties.get("format") : null;
+
+      // Check if tableType or format contain "iceberg"
+      boolean isIcebergByType = tableType != null && tableType.toLowerCase(Locale.ROOT).contains("iceberg");
```
Comment (jmnunezizu) [inline beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/handler/ExpiredMetadataHandler.java:168]: Why do you need to use `Locale.ROOT`? Isn't `.toLowerCase()` enough in this case?

File: beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/handler/ExpiredMetadataHandler.java [hunk 3]
```diff
@@ -146,6 +158,30 @@ private boolean cleanupPartition(
...
+      boolean isIcebergByFormat = format != null && format.toLowerCase(Locale.ROOT).contains("iceberg");
```
Comment (jmnunezizu) [inline beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/handler/ExpiredMetadataHandler.java:169]: Same as above.

File: beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/filter/IcebergTableListenerEventFilter.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.Locale;
+import java.util.Map;
+
+
+public class IcebergTableListenerEventFilter implements ListenerEventFilter {
+
+  private static final Logger log = LogManager.getLogger(IcebergTableListenerEventFilter.class);
+
+  private static final String FORMAT_KEY = "format";
+  private static final String FORMAT_ICEBERG_VALUE = "ICEBERG/PARQUET";
```
Comment (jmnunezizu) [inline beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/filter/IcebergTableListenerEventFilter.java:34]: What if it's an Iceberg table with a different file format? 
All Iceberg tables should be filtered, regardless of their file format.

**Review by jmnunezizu — CHANGES_REQUESTED** [review]

File: beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/handler/ExpiredMetadataHandler.java (hunk 2 above)
Comment (HamzaJugon) [reply beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/handler/ExpiredMetadataHandler.java:168]: yes not needed, updated.

File: beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/filter/IcebergTableListenerEventFilter.java (hunk 4 above)
Comment (HamzaJugon) [reply beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/filter/IcebergTableListenerEventFilter.java:34]: ah yes, updated filter to look for 'iceberg' only in table format.

**Pull request closed without merging**